frontend/src/metabase/querying/filters/components/DatePicker/DatePicker.stories.tsx

```diff
@@ -0,0 +1,230 @@
+    } else {
+      document.documentElement.setAttribute("data-metabase-theme", "light");
+    }
+  }, [isDarkTheme]);
+
+  return (
+    <>
+      <Popover opened position="bottom-start" withinPortal={false}>
+        <Popover.Target>
+          <div></div>
+        </Popover.Target>
+        <Popover.Dropdown>
+          <DatePicker {...args} />
+        </Popover.Dropdown>
+      </Popover>
+      {args.snapshotSize && (
+        <Box
+          pos="absolute"
+          // the space between the popover and the target element
+          mt="8px"
+          w={args.snapshotSize.width}
+          h={args.snapshotSize.height}
+        />
+      )}
+    </>
+  );
+};
+
+export const AllOptions = {
+  render: Template,
+  play: async ({ canvasElement }: { canvasElement: HTMLCanvasElement }) => {
+    const canvas = within(canvasElement);
+    const yesterday = canvas.getByRole("button", {
+      name: "Yesterday",
+    });
+    yesterday.classList.add("pseudo-hover");
+  },
+};
+export const AllOptionsDarkTheme = merge(AllOptions, {
+  args: {
+    theme: "dark",
+  },
+});
+
+export const Exclude = {
+  render: Template,
+  play: async ({ canvasElement }: { canvasElement: HTMLCanvasElement }) => {
+    const canvas = within(canvasElement);
+    await userEvent.click(
+      canvas.getByRole("button", {
+        name: "Exclude…",
+      }),
+    );
+    canvas
+      .getByRole("button", { name: "Days of the week…" })
+      .classList.add("pseudo-hover");
+  },
+};
+export const ExcludeDarkTheme = merge(Exclude, {
+  args: {
+    theme: "dark",
+  },
+});
+
+export const ExcludeDayOfWeek = {
+  render: Template,
+  args: {
+    value: {
+      type: "exclude",
+      operator: "!=",
+      unit: "day-of-week",
+      values: [3, 4, 5],
+    },
+  },
+};
+export const ExcludeDayOfWeekDarkTheme = merge(ExcludeDayOfWeek, {
+  args: {
+    theme: "dark",
+  },
+});
+
+export const RelativeCurrent = {
+  render: Template,
+  args: {
+    value: { type: "relative", unit: "day", value: "current" },
+  },
+  play: async ({ canvasElement }: { canvasElement: HTMLCanvasElement }) => {
+    const canvas = within(canvasElement);
+    const week = canvas.getByRole("button", {
+      name: "Week",
+    });
+    week.classList.add("pseudo-hover");
+    const next = canvas.getByRole("tab", {
+      name: "Next",
+    });
+    next.classList.add("pseudo-hover");
+  },
+};
+export const RelativeCurrentDarkTheme = merge(RelativeCurrent, {
+  args: {
+    theme: "dark",
+  },
+});
+
+export const RelativePrevious = {
+  render: Template,
+  args: {
+    value: {
+      options: { includeCurrent: true },
+      type: "relative",
+      unit: "month",
+      value: -12,
+    },
+  },
+  play: async ({ canvasElement }: { canvasElement: HTMLCanvasElement }) => {
+    const canvas = within(canvasElement);
+    const next = canvas.getByRole("tab", {
+      name: "Next",
+    });
+    next.classList.add("pseudo-hover");
+  },
+  decorators: [withMockDate],
```

Comment:
For relative date widget, we need to mock date, otherwise the snapshot will change everyday 😆